Docs: enhance doc for Solaris icons (#2075)
Orange-OpenSource/Orange-Boosted-Bootstrap: site/content/docs/5.3/extend/icons.md

While Boosted doesn't include an icon set by default, Orange does have its own comprehensive icon library called Solaris. While most icon sets include multiple file formats, we prefer SVG implementations for their improved accessibility and vector support.

> [Solaris icons library]({{< param icons >}}) is a growing library of icons that are designed by [Orange's Group Design Team](https://system.design.orange.com/).

Reviewer: ```suggestion
[Solaris icon library]({{< param icons >}}) is a growing library of icons that are designed by [Orange Design System design team](https://system.design.orange.com/).
```

It features an icon search tool based on categories and keywords, and allows icons to be downloaded in various formats for use in design and development.

> Please refer to the Solaris Icon Library user documentation for full design and development guidelines.

Reviewer: There is probably a missing link here

> They are not open-source though and should only be used for Orange branded projects. Please refer to the [icons license file]({{< param icons_license >}}) for legal information.

Reviewer: ```suggestion
They are not open-source though and should only be used for Orange projects. Please refer to the [icons license file]({{< param icons_license >}}) for legal information.
```

## Use Solaris icons

There are many ways to use Solaris icons in a web page:
- Bundled, if you have several icons in your project:
  - [SVG sprite]({{< docsref "extend/icons#svg-sprite" >}}) (preferred choice for flexibility, performance and accessibility)
  - [Web font]({{< docsref "extend/icons#web-font" >}})
- Individually, if you have only few icons to render:
  - [Inline SVG]({{< docsref "extend/icons#inline-svg" >}})
  - [SVG external image]({{< docsref "extend/icons#svg-external-image" >}})
  - [CSS background SVG]({{< docsref "extend/icons#css-background-svg" >}})

> See Solaris icons library [develop documentation]({{< param icons_doc >}}) for complete information about the different formats and their pros and cons.

Reviewer: The link doesn't go to the right page. Plus, we could say "development" instead of "develop"; it sounds weird to me.

All icons can be retrieved easily in any of these formats in the Solaris icons finder, except for the specific case of [warning icon]({{< docsref "extend/icons#warning-icon" >}}) which can be copied below.

> ### SVG Sprite

Reviewer: ```suggestion
### SVG sprite

```

This technique is the preferred choice for flexibility, performance and accessibility.

Using the [Solaris icons finder]({{< param icons >}}), you can generate an SVG sprite—a single SVG file containing all your icons—and insert an icon through the `<use>` element.

> This is similar to an `<img>` element, but with the power of `currentColor` for easy theming: see in this example how some icons inherit their color from the parent's light or dark theme, whereas some others get their color from [text Orange's colors utilities]({{< docsref "/utilities/colors" >}}) or local style.

Reviewer: "text Orange's colors utilities" doesn't go directly to the corresponding section.

While Boosted doesn't include an icon set by default, Orange does have its own comprehensive icon library called Solaris. While most icon sets include multiple file formats, we prefer SVG implementations for their improved accessibility and vector support.

> [Solaris icons library]({{< param icons >}}) is a growing library of icons that are designed by [Orange's Group Design Team](https://system.design.orange.com/).

Reviewer: ```suggestion
[Solaris icons library]({{< param icons >}}) is a growing library of icons that are designed by [Orange Design System design team](https://system.design.orange.com/).
```

Using the Solaris icons finder, you can generate a Web font containing all your icons and use it like a classic font.
Icons can be styled through CSS properties like `font-size` and `color`.

### Inline SVG
This technique should only be used if you have a few icons to render, and not rendering the same icon multiple times.

You can embed your icons directly within the HTML of your page (as opposed to an external image file).

{{< example class="mt-0" >}}
<svg fill="currentColor" width="2em" height="2em" viewBox="0 0 1000 1000" aria-hidden="true" focusable="false">
  <path d="M656.7 422.409a229.96 229.96 0 0 1-315.39.008A224.95 224.95 0 0 0 224.064 615H224v210a100 100 0 0 0 100 100h450V620a224.94 224.94 0 0 0-117.3-197.591M679 255A180 180 0 1 1 499 75a180 180 0 0 1 180 180" style="fill-rule:evenodd"/>
</svg>
{{< /example >}}

The `fill="currentColor"` attribute is required if you want to change the color of the icons on the fly, and that the foreground color changes in dark/light mode.

### SVG external image

You can use it when:
- you only have a few icons to render

> - you don't need to change the style or color of the icon

Author: discussed with LM: seems too complicated to explain. Advanced users  should be able to do it by themselves

You can download the Solaris icons SVGs from the Solaris icons finder, copy them into your project and reference them like normal images with the `<img>` element.

{{< example class="mt-0" >}}
<img src="/docs/{{< param docs_version >}}/assets/img/boosted-cross.svg" alt="" width="32" height="32">
{{< /example >}}

> ### CSS background SVG

Author: I change and use css var to switch color with theme

Then, you can:
- use [text color utilities classes]({{< docsref "utilities/colors" >}}) like `.text-success` that change current color value
- specify a color using CSS variables like: `style="color: var(--bs-success)"`

## Icons accessibility

{{< callout info >}}
For more details, **Orange Accessibility Guidelines** provides [a deep-dive article regarding SVG accessibility](https://a11y-guidelines.orange.com/en/articles/accessible-svg/).
{{< /callout >}}

#### Decorative icons
Purely **decorative icons** (like repeating information of an adjacent text) must be hidden to assistive technologies:
- for `<svg>` tag, use the attributes `aria-hidden="true"` and `focusable="false"`.
- for `<span>` tag, use the attribute `aria-hidden="true"`
- for `<img>`, use an empty `alt` attribute
- CSS background images are intended to be decorative

#### Informative/meaningful icons
If the icon is **meaningful**, e.g. only content of a button, you have to provide an appropriate alternative text: for example, the description of the icon or the description of the action triggered.

> The best way to do this is to keep the icon hidden to assistive technologies (see above) and add a visually hidden label (which will be perceived by assistive technologies).

Author: yes you're right

</div>

## About SVGs accessibility

{{< callout warning >}}
SVGs are awesome to work with, but they do have some known quirks to work around. Given the numerous ways in which SVGs can be used, **we haven’t systematically included these attributes and workarounds in our code**.
{{< /callout >}}

Known issues include:

* On decorative images:
  * **Focus handling is broken in Edge Legacy.** When embedding your SVGs, add `focusable="false"` to the `<svg>` element.
* On informative images:
  * **Browsers inconsistently announce `<svg>` tags as images with voice assistance.** Include `role="img"` to avoid any issues.
  * **Safari skips `aria-label` when using non-focusable SVGs.** Prefer using `aria-labelledby` referencing a `title` element inside `svg` tag.

For more details, **Orange Accessibility Guidelines** provides [a deep-dive article regarding SVG accessibility](https://a11y-guidelines.orange.com/en/articles/accessible-svg/).

## Create a SVG sprite file

To create your own SVG sprite file, containing only the icons you need:

> - download the icons needed from [ODS website](https://system.design.orange.com/0c1af118d/p/939811-solaris-icon-library/b/795c81)

Reviewer 2: Url seems to be broken, here is the good one atm: https://system.design.orange.com/0c1af118d/p/65c68d-solaris-icon-library/b/639b3c

Another approach would consist of describing the way to have access to the good link maybe ? (For instance the good link is at the very bottom of the page)

Maybe the good approach is to do as in the `Learn more about Solaris` link at the top of this page ?

Since this Url will certainly change in the future, I don't really know what's the good approach on this here.

Author: URL shouldn't change, even with the new icons finder. I think @julien-deramond checks DSM URLs time to time to be sure they haven't change?

> - download the icons needed from [ODS website](https://system.design.orange.com/0c1af118d/p/939811-solaris-icon-library/b/795c81)

Reviewer 2: I don't know if it's feasible with markdown but maybe an ordered list would be better here.
